aptos-move/framework/aptos-stdlib/sources/cryptography/groups.move

```diff
@@ -0,0 +1,749 @@
+        assert!(x"3c8dd3f68a360f9c5ba81fad2be3408bdc3070619bc7bf3794851bd623685a5036ef5f1388c0541e58c3d2b2dbd19c04c83472247446b1bdd44416ad1c1f929a3f01ed345be35b9b4ba20f17ccf2b5208e3dec8380d6b8c337ed31bff673020dddcc1399cdf852dab1e2c8dc3b0ce819362f3a12da56f37aee93d3881ca760e467942c92428864a6172c80bf4daeb7082070fa8e8937746ae82d57ec8b639977f8ceaef21a11375de52b02e145dc39021bf4cab7eeaa955688a1b75436f9ec05" == serialize_element_uncompressed(&point_7g_calc), 1);
+        assert!(x"3c8dd3f68a360f9c5ba81fad2be3408bdc3070619bc7bf3794851bd623685a5036ef5f1388c0541e58c3d2b2dbd19c04c83472247446b1bdd44416ad1c1f929a3f01ed345be35b9b4ba20f17ccf2b5208e3dec8380d6b8c337ed31bff673020d" == serialize_element_compressed(&point_7g_calc), 1);
+
+        // Point double.
+        let point_2g = element_scalar_mul(&scalar_2, &point_g);
+        let point_double_g = element_double(&point_g);
+        assert!(element_eq(&point_2g, &point_double_g), 1);
+
+        // Point negation.
+        let point_minus_7g_calc = element_neg(&point_7g_calc);
+        assert!(x"3c8dd3f68a360f9c5ba81fad2be3408bdc3070619bc7bf3794851bd623685a5036ef5f1388c0541e58c3d2b2dbd19c04c83472247446b1bdd44416ad1c1f929a3f01ed345be35b9b4ba20f17ccf2b5208e3dec8380d6b8c337ed31bff673028d" == serialize_element_compressed(&point_minus_7g_calc), 1);
+        assert!(x"3c8dd3f68a360f9c5ba81fad2be3408bdc3070619bc7bf3794851bd623685a5036ef5f1388c0541e58c3d2b2dbd19c04c83472247446b1bdd44416ad1c1f929a3f01ed345be35b9b4ba20f17ccf2b5208e3dec8380d6b8c337ed31bff673020dceddeb663207acdf4d1d8bd4c2f3c304eec676e4c67b3decd07eb16a68a416806f181fb1731fb7a482baff799c6349118b3a057176c88a4f17d2fcc4729c12a72b27020486c1f909dae682123f6f3d62bcb8808bc7fc85f41145c8e4b3181414" == serialize_element_uncompressed(&point_minus_7g_calc), 1);
+
+        // Point addition.
+        let point_9g = element_scalar_mul(&scalar_9, &point_g);
+        let point_2g = element_scalar_mul(&scalar_2, &point_g);
+        let point_2g_calc = element_add(&point_minus_7g_calc, &point_9g);
+        assert!(element_eq(&point_2g, &point_2g_calc), 1);
+
+        // Multi-scalar multiplication.
+        let point_14g = element_scalar_mul(&scalar_from_u64<BLS12_381_G2>(14), &point_g);
+        let scalar_1 = scalar_from_u64<BLS12_381_G2>(1);
+        let scalar_2 = scalar_from_u64<BLS12_381_G2>(2);
+        let scalar_3 = scalar_from_u64<BLS12_381_G2>(3);
+        let point_2g = element_scalar_mul(&scalar_2, &point_g);
+        let point_3g = element_scalar_mul(&scalar_3, &point_g);
+        let scalars = vector[scalar_1, scalar_2, scalar_3];
+        let points = vector[point_g, point_2g, point_3g];
+        let point_14g_calc = element_multi_scalar_mul(&scalars, &points);
+        assert!(element_eq(&point_14g, &point_14g_calc), 1);
+
+        // Hash to group.
+        let point = hash_to_element<BLS12_381_G2>(x"1234");
+        assert!(element_eq(&group_identity<BLS12_381_G2>(), &element_scalar_mul(&scalar_from_u64<BLS12_381_G2>(0), &point)), 1);
```

Comment:
Uhm, what does this achieve?

Reply:
Let's move this feature into a separate PR!